torch/_meta_registrations.py

```diff
@@ -466,6 +466,22 @@ def is_fast_path(src, scale, output, padding_idx):
     return output, offset2bag, bag_size, max_indices
 
 
+@torch.library.impl(meta_lib, "diag.out")
+def meta_diag_out(self, dim=0, *, out):
+    assert self.dim() in (1, 2), "matrix or a vector expected"
+    if self.dim() == 1:
+        sz = self.size(0) + abs(dim)
+        torch._resize_output_(out, (sz, sz), self.device)
+        return out
+    else:
+        if dim >= 0:
+            sz = min(self.size(0), self.size(1) - dim)
+        else:
+            sz = min(self.size(0) + dim, self.size(1))
+        torch._resize_output_(out, (sz,), self.device)
+        return out
```

Comment:
It's more idiomatic to use the `out_wrapper` so you don't have to manually deal with the out parameter (and this raises the question how come the functional version doesn't need a meta registration.)